src/frontend/components/DashboardVideoLiveControlPane/widgets/DashboardVideoLiveWidgetGeneralTitle/index.spec.tsx

```diff
@@ -0,0 +1,292 @@
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import fetchMock from 'fetch-mock';
+import MatchMediaMock from 'jest-matchmedia-mock';
+import React from 'react';
+import { Toaster } from 'react-hot-toast';
+import { QueryClient, QueryClientProvider, setLogger } from 'react-query';
+
+import { report } from 'utils/errors/report';
+import { videoMockFactory } from 'utils/tests/factories';
+import { wrapInIntlProvider } from 'utils/tests/intl';
+import { DashboardVideoLiveWidgetGeneralTitle } from '.';
+
+let matchMedia: MatchMediaMock;
+
+jest.mock('data/appData', () => ({
+  appData: {
+    resource: {
+      id: '1',
+    },
+  },
+}));
+
+jest.mock('utils/errors/report', () => ({
+  report: jest.fn(),
+}));
+
+setLogger({
+  log: () => {},
+  warn: () => {},
+  error: () => {},
+});
+
+describe('test toggle button in <DashboardVideoLiveWidgetGeneralTitle />', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    matchMedia = new MatchMediaMock();
+  });
+  afterEach(() => {
+    fetchMock.restore();
+    matchMedia.clear();
+  });
+
+  it('clicks on the toggle button for activate the live recording', async () => {
+    const mockedVideo = videoMockFactory({
+      id: 'video_id',
+      allow_recording: false,
+    });
+
+    fetchMock.patch('/api/videos/video_id/', {
+      ...mockedVideo,
+      allow_recording: true,
+    });
+    const queryClient = new QueryClient();
+
+    render(
+      wrapInIntlProvider(
+        <QueryClientProvider client={queryClient}>
+          <DashboardVideoLiveWidgetGeneralTitle video={mockedVideo} />,
+        </QueryClientProvider>,
+      ),
+    );
+
+    screen.getByText('Activate live recording');
+    const recordingToggleButton = screen.getByRole('checkbox', {
+      name: 'Activate live recording',
+    });
+
+    await act(async () => userEvent.click(recordingToggleButton));
+    expect(fetchMock.calls()).toHaveLength(1);
+    expect(fetchMock.lastCall()![0]).toEqual('/api/videos/video_id/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'PATCH',
+      body: JSON.stringify({
+        allow_recording: true,
+      }),
+    });
+    expect(report).not.toHaveBeenCalled();
```

Comment:
Can't we test visually it has been updated? I've seen you had planned on using a success message but it's not used in your file.

Reply:
The success message is now used, and tests are now checking the toast is well displayed.